src/status_im/protocol/web3/inbox.cljs

```diff
@@ -44,14 +44,40 @@
                                                   (swap! peers conj enode)
                                                   (success-fn result))))))
 
-(defn mark-trusted-peer [web3 enode success-fn error-fn]
+(defn extract-enode-id [enode]
+  (-> enode
+      env.utils/get-host
+      (string/split "@")
+      (get 0)))
```

Comment:
How does this deal with badly formatted input/nils?

Reply:
It handles it well bc of clojure nil punning. But i've added more test cases for that.